packages/plugin-meta-extractor/README.md

```diff
@@ -0,0 +1,92 @@
+# Grafana / Plugin Meta Extractor
+
+`@grafana/plugin-meta-extractor` is a small package that can be used to extract meta-information from the source code of a Grafana plugin. The main use case is to be able to automatically generate static information for a plugin without needing to manually edit the meta-info files (which might get forgotten).
+
+## Install
+
+```
+npm install @grafana/plugin-meta-extractor
+```
+
+## Usage
+
+#### With Webpack
+
+The package exposes a webpack plugin that can be used to generate plugin meta-information on every build based on the source code.
+
+```ts
+// webpack.config.ts
+// -----------------
+
+import { GrafanaPluginMetaExtractor } from '@grafana/plugin-meta-extractor';
+
+export default {
+  // ...
+  plugins: [
+    // Creates a "plugin.generated.json" file in the output directory
+    // `entryFile` - an absolute path pointing to the module.(ts|tsx) file of your plugin
```

Comment:
After having a look I think it would be tricky to completely rely on the webpack config inside the plugin.

`compiler.options.entry` would give us the actual entry points, however based on it's types it can also be a function that returns a promise, or an object with different entry points (which is based on how the user writes the webpack config). We could say that we create a contract between our webpack config and this plugin, however I am not sure if that's the best thing to do. I would probably rather keep the argument to be more explicit and reusable.